jetpack-beta.php

```diff
@@ -1115,6 +1117,38 @@ static function check_for_main_file( $source, $remote_source ) {
 
 		return $source;
 	}
+
+	/**
+	 * Custom error handler to intercept errors and log them using Jetpack's own logger.
+	 *
+	 * @param int $errno error code.
+	 * @param string $errstr error message.
+	 * @param string $errfile file name where the error happened.
+	 * @param int $errline line in the code.
+	 *
+	 * @return bool whether to make the default handler handle the error as well.
+	 */
+	public static function custom_error_handler( $errno, $errstr, $errfile, $errline ) {
+
+		if ( class_exists( 'Jetpack' ) && method_exists( 'Jetpack', 'log' ) ) {
+			$error_string = sprintf( "%s, %s:%d", $errstr, $errfile, $errline );
```

Comment:
I wonder if we should check whether `$errfile` includes `jetpack`, and only save to Jetpack Log if the error is about a Jetpack file. This way we'll avoid populating the log with many entries that may not be related to Jetpack at all.